Great! Just to check though: does this address the warnings that were being generated during recent builds, e.g.:
```
[1075/1118] Building CXX object tools/mlir/python/CMakeFiles/MLIRPythonModules.extension._mlir.dso.dir/__/lib/Bindings/Python/IRCore.cpp.o
/data/nfs_home/rmorel/llvm-project.git/elementwise-maximumf/mlir/lib/Bindings/Python/IRCore.cpp:2813:11: warning: unused function '_Py_XNewRef' [-Wunused-function]
PyObject *_Py_XNewRef(PyObject *obj) {
          ^
/data/nfs_home/rmorel/llvm-project.git/elementwise-maximumf/mlir/lib/Bindings/Python/IRCore.cpp:2822:11: warning: unused function '_Py_NewRef' [-Wunused-function]
PyObject *_Py_NewRef(PyObject *obj) {
          ^
2 warnings generated.
```

I think those are probably related to the same PR (or follow-up fixes).

--- a/mlir/cmake/modules/AddMLIRPython.cmake
+++ b/mlir/cmake/modules/AddMLIRPython.cmake
@@ -704,7 +704,12 @@ function(add_mlir_python_extension libname extname)
       # NanobindAdaptors.h uses PyClassMethod_New to build `pure_subclass`es but nanobind
       # doesn't declare this API as undefined in its linker flags. So we need to declare it as such
       # for downstream users that do not do something like `-undefined dynamic_lookup`.
-      target_link_options(${libname} PUBLIC "LINKER:-U,_PyClassMethod_New")
+      # Same for the rest.
+      target_link_options(${libname} PUBLIC
+        "LINKER:-U,_PyClassMethod_New"
+        "LINKER:-U,_PyCode_Addr2Location"
+        "LINKER:-U,_PyFrame_GetLasti"
+      )
     endif()
   endif()
 

--- a/mlir/lib/Bindings/Python/IRCore.cpp
+++ b/mlir/lib/Bindings/Python/IRCore.cpp
@@ -2810,7 +2810,7 @@ private:
 
 // bpo-42262 added Py_XNewRef()
 #if !defined(Py_XNewRef)
-PyObject *_Py_XNewRef(PyObject *obj) {
+[[maybe_unused]] PyObject *_Py_XNewRef(PyObject *obj) {
   Py_XINCREF(obj);
   return obj;
 }
@@ -2819,7 +2819,7 @@ PyObject *_Py_XNewRef(PyObject *obj) {
 
 // bpo-42262 added Py_NewRef()
 #if !defined(Py_NewRef)
-PyObject *_Py_NewRef(PyObject *obj) {
+[[maybe_unused]] PyObject *_Py_NewRef(PyObject *obj) {
   Py_INCREF(obj);
   return obj;
 }

